FFMpegCore/FFMPEG/FFMpeg.cs

```diff
@@ -108,7 +109,10 @@ public Bitmap Snapshot(VideoInfo source, FileInfo output, Size? size = null, Tim
 
             if (output.Exists)
             {
-                output.Delete();
+                if (!persistSnapshotOnFileSystem)
```

Comment:
Please reduce the nesting by adding this check in the first `if` statement. It's good to merge afterwards.